### PR TITLE
Streamline Linode extension

### PR DESCRIPTION
**Pull request by johnwcallahan**
Centralizes and simplifies the way we "extend" Linodes. Before, we had several interfaces called `ExtendedLinode`, and `LinodeWithMaintenance`, and `LinodeWithMaintenanceAndDisplayStatus`, etc. They were used inconsistently and in some places incorrectly.

 Now there are two options:

`ShallowExtendedLinode` (Linode + maintenance/status/event info)
`DeepExtendedLinode` (ShallowExtendedLinode + configs, disks, volumes, etc.)

The fundamental difference is that `ShallowExtendedLinode` requires top-level requests only to fulfill the interface (notifications, events, and eventually VLANs) while `DeepExtendedLinode` requires child requests for each Linode (Linode configs, disks, and eventually network interfaces).

So:
- `ShallowExtendedLinode` == good for Linode Landing
- `DeepExtendedLinode` == good for Linode Details

To do: tests, documentation.

### CODE REVIEW

File: packages/manager/src/containers/withShallowExtendedLinodes.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { connect } from 'react-redux';
```
Comment (Jskobos) [inline packages/manager/src/containers/withShallowExtendedLinodes.ts:1]: This is like a sub-slice of withLinodes.container, I think? And I can't see where it's used.

File: packages/manager/src/store/linodes/types.ts [hunk 2]
```diff
@@ -1,9 +1,37 @@
-import { LinodeWithMaintenance as L } from './linodes.helpers';
+import { LinodeStatus, Linode } from '@linode/api-v4/lib/linodes';
+import { ExtendedEvent } from 'src/store/events/event.types';
 
-/* tslint:disable-next-line */
-export interface LinodeWithMaintenance extends L {}
+export interface LinodeWithMaintenance extends Linode {
```
Comment (Jskobos) [inline packages/manager/src/store/linodes/types.ts:4]: Maybe don't export these, if the idea is to have just plain Linodes with 2 and only 2 extensions

**Review by Jskobos — APPROVED** [review]
Makes sense, I left some cleanup stuff but everything seems in order.

**Review by WilkinsKa1 — APPROVED** [review]
This looks good, thanks for streamlining these. I was never quite sure which interface to use but this delineation is much clearer 👍

**Pull request merged**